### PR TITLE
Testing for Signup, Login and Video Search Use Cases

### PR DESCRIPTION
**Pull request by caratooo**
The pull request includes tests for the Signup, Login and Video Search Use Cases.

All tests have 100% coverage except all the Views.

### CODE REVIEW

File: src/data_access/VideoSearchDataAccessObject.java
Comment (jonathanjschen913) [inline src/data_access/VideoSearchDataAccessObject.java:1]: I think we can consider naming this more explicitly as a testing DAO after the merge, but that's more of a stylistic suggestion. Nicely done!

File: test/entities/VideoEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                "FIE Fencing Channel",
+                "Grand Prix Turin 2020 Women's Foil Final THIBUS FRA vs KIEFER USA",
+                "",
+                dateTime1,
+                12996,
+                169,
+                9);
+
+        DateTime dateTime2 = new DateTime("2020-02-19T18:48:38.000Z");
+        longVideo = new Video("QbwgQSwMSGM",
```
Comment (jonathanjschen913) [inline test/entities/VideoEntityTest.java:28]: Sorry if I'm not following here, but is this longVideo expected to be used?

File: test/use_case/video_search/VideoSearchInteractorTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            @Override
+            public void prepareFailView(String error) {
+                fail("This use case should not fail.");
+            }
+
+        };
+
+        // interactor + execute
+        VideoSearchInputBoundary interactor = new VideoSearchInteractor(youtubeRepository, historyRepository, presenter);
+        interactor.execute(inputData);
```
Comment (jonathanjschen913) [inline test/use_case/video_search/VideoSearchInteractorTest.java:53]: Does this account for the video growth in statistics like view count, like count, and comment count?

File: test/entities/VideoEntityTest.java (hunk 1 above)
Comment (caratooo) [reply test/entities/VideoEntityTest.java:28]: Oh my bad, it should be used in the checkDescriptionLength() method - nice catch!

**Pull request merged**